src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -319,15 +319,16 @@ private void moveElementDirectory(UUID elementUuid, UUID newDirectoryUuid, Strin
 
         DirectoryElementEntity oldDirectory = repositoryService.getElementEntity(element.getParentId()).orElseThrow();
         updateElementParentDirectory(element, newDirectoryUuid);
```

Comment:
i think something is missing here.
when a directory is moved we need to find all the element in this directory and element in his sub dir and reindex the in elastic search.
you can check that the behavior is not correct by moving a dir that contains some elements , and making a search in gridexplore